loki/transform/transform_loop.py

```diff
@@ -654,7 +654,7 @@ class FissionTransformer(NestedMaskedTransformer):
 
     def __init__(self, loop_pragmas, active=True, **kwargs):
         super().__init__(active=active, require_all_start=True, greedy_stop=True, **kwargs)
-        self.loop_pragmas = loop_pragmas
+        self.loop_pragmas = dict((k, v) for k, v in loop_pragmas.items())
```

Comment:
Why is that required? And would `self.loop_pragmas = loop.pragmas.copy()` achieve the same?

Reply:
dict lookups kept failing, even though the keys were definitely in the dict. It was a strange bug; I could get it to pass by printing the keys before the lookup. This seems to have solved the problem.

Reply:
No, the bug reappears if I use `loop_pragmas.copy()`.

Reply:
This really does sound odd. Any chance we can figure out why such a seemingly unrelated change "fixes" things?

Reply:
The `DataflowAnalysisAttacher/Detacher` by default had `invalidate_source` set to true. And now that the expression nodes no longer have a source, calls to `dataflow_analysis_attached` were invalidating source. This was the source (pun-intended) of those funky dict lookup failures.

I think the `DataflowAnalysis` should not modify the nodes, since it is an analysis pass. So I'm setting `invalidate_source` to false in the constructor.